plugins/scaffolder/src/components/TaskPage/TaskPage.tsx

```diff
@@ -266,6 +273,26 @@ export const TaskPage = () => {
 
   const { output } = taskStream;
 
+  const handleEditInputs = () => {
+    if (!taskStream.task || !taskStream.task?.spec.metadata?.name) {
+      navigate(generatePath(rootLink()));
+    }
+
+    const formData =
+      taskStream.task!.spec.apiVersion === 'backstage.io/v1beta2'
+        ? taskStream.task!.spec.values
+        : taskStream.task!.spec.parameters;
+
+    navigate(
+      generatePath(
+        `${rootLink()}/templates/:templateName?${qs.stringify({ formData })}`,
```

Comment:
To shorten things and reduce the noise, what do you think about perhaps using local storage for this instead? If we serialize the form state into local storage, and then just have a single query parameter `?restart=true` or whatnot on the form page, that decides whether to even try loading from there. If it isn't set, the page can wipe that local storage key as well.

Reply:
hm interesting ... 

I guess another use-case I should've mentioned I am hoping to achieve with this change is to also support generating and sharing pre-filled template page URLs externally. 

For example, we're trying to ease onboarding users to register existing entities they own by reading their source data from an external system and pre-filling them into a custom "register new componment" template (similar to the "Open PR to auto create your entity file" functionality but with extra required fields). We do it this way since there is some extra metadata we can't deduce automatically and rather not auto ingest entities with incomplete data (quality over quantity) so we would like to construct custom template URLs we can send to their respective owners to allow them to open the pre-filled template, fill in the remaining missing info, and submit it for registration.

Passing that info through query params makes this a lot easier. This will also make it work the same way you can share catalog URLs with filters pre-set through query params.

Thoughts?

Reply:
Alright, that makes a lot of sense! Interesting, that sounds useful. All good by me, @benjdlambert just wants to have a look as well, maybe tomorrow.

Reply:
Had a thought about this yesterday a little bit @kuangp, how the sharing of the URL's sounds like an interesting use case, how do you expect to generate these URLs? Like do you fill in half the form and then it update in the URL bar or do you just use a one time script that can generate some autopopulated values?

I'm thinking that it would be good to maybe try an load these values in from localStorage too, I haven't tested this yet, but does support navigating back using the browser over the button? Thinking that we probably want to support that use case too that if you navigate back from a running job it persists the values that were in the form already if that makes sense rather than always remembering to click the button? That do you think @kuangp? 🙏

Reply:
@benjdlambert for our use case, we have an automated script that will source data from another system and generate pre-filled template URLs according to the current schema of our template parameters.

For supporting native browser back navigation I added a [change](https://github.com/backstage/backstage/pull/8504/files#diff-119953a802a83be55cdaa1572e7f31744823fb666003c9b21df47df4256196a2R148) to also encode the form state into the history state prior to navigating to the task page. This will support the scenario where a user going through the common workflow of filling out the template page -> trigger run -> task page can use either the 'Start Over' button or just hit Back on their browser. The 'Start Over' button however will support future use-cases where a user may be able to view historical runs/check back on a long running task and will not directly originate from the template page. 

Personally, I'm hesitant about the `localStorage` implementation since it feels like we would be abusing the semantics a bit to handle stateful navigations/data passing whereas `localStorage` should be meant more for persistent cross-session storage eg. user preferences.  

Let me know if you disagree.

Reply:
Just tried this out and it works great, awesome job. Think the experience is great and exactly what we want, but just wanna confirm with @Rugvip especially that how we're routing to the other pages is the way we want to encourage when navigating within a plugin 🙏  

A little side thought is that the URL that we're sharing around isn't versioned or anything, so over time these URL's might not be valid with different versions of the scaffolder frontend, let's say if we change the param name or whatever, but I think that's fine for now at least. it's something that might change in the future, but easy to support both the new and the old and deprecate the old anyway.

And yeah, I understand what you're saying about `localStorage`. It is maybe not the best place to store it, but it would make the URL's clean after navigating to the link as we could just write the data to `localStorage` and also we would be able to write state to the `localStorage` state in realtime rather than it writing after you click `create`. 

But to be honest right now I'm not sure that either of those reasons are dealbreaker right now, so I'm happy to keep this this state in the query right now, on the understanding that it's not a stable API right now and could change in future releases whatever we decide to do moving forward.

Reply:
sounds reasonable, thanks!